### ISSUE TITLE
CI failing on PySide6

### ISSUE THREAD

**Issue by mdickinson** [enthought/traitsui#1892]
The main (EDM-based) continuous integration job is currently failing with PySide6 on Ubuntu. An example failure is here: https://github.com/enthought/traitsui/runs/6628648187?check_suite_focus=true

```
======================================================================
ERROR: traitsui (unittest.loader._FailedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/traitsui/__init__.py", line 76, in load_tests
    from traitsui.tests._tools import filter_tests
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/traitsui/tests/_tools.py", line 16, in <module>
    from pyface.toolkit import toolkit_object
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/toolkit.py", line [23](https://github.com/enthought/traitsui/runs/6628648187?check_suite_focus=true#step:8:24), in <module>
    toolkit = toolkit_object = find_toolkit("pyface.toolkits")
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/base_toolkit.py", line [26](https://github.com/enthought/traitsui/runs/6628648187?check_suite_focus=true#step:8:27)3, in find_toolkit
    return import_toolkit(ETSConfig.toolkit, entry_point)
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/base_toolkit.py", line 2[29](https://github.com/enthought/traitsui/runs/6628648187?check_suite_focus=true#step:8:30), in import_toolkit
    raise RuntimeError(msg)
RuntimeError: No pyface.toolkits plugin could be loaded for qt
```

**Comment by mdickinson** [comment]
Relevant traceback:
```
DEBUG:pyface.base_toolkit:could not import module 'PySide6.QtCore'
Traceback (most recent call last):
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/base_toolkit.py", line 218, in import_toolkit
    toolkit_object = plugin.load()
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/importlib_metadata/__init__.py", line 167, in load
    module = import_module(match.group('module'))
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/ui/qt4/init.py", line 17, in <module>
    from pyface.qt import QtGui
  File "/home/runner/work/traitsui/traitsui/.edm/envs/traitsui-test-3.6-pyside6/lib/python3.6/site-packages/pyface/qt/QtGui.py", line 57, in <module>
    from PySide6.QtGui import *
ImportError: could not import module 'PySide6.QtCore'
could not import module 'PySide6.QtCore'
```

**Comment by mdickinson** [comment]
Fixed in #1893 (see also #1898).
